src/components/Input/Input.tsx

```diff
@@ -12,7 +14,12 @@ const Input = forwardRef<HTMLDivElement, InputWrapperProps & InputProps>((props,
     const { marginProps, restProps: withoutMargin } = extractWrapperMarginProps(withoutClassName);
     const { widthProps, restProps } = extractWidthProps(withoutMargin);
 
-    const { label, onChange, size, id, ...rest } = restProps;
+    const { label, onChange, size, ...rest } = restProps;
+    const id = useMemo(() => {
```

Comment:
This will probably repeat across several instances of different components. Makes sense to have a `useGeneratedId(props.id)` hook that we can reuse everywhere?

Reply:
I see that the code is duplicated, but if I want to extract it, I will have to make it generic enough for reuse. The way I see the generic makes it complex because we need to provide an option to avoid id generation (see the use of `label`):

```
const useGeneratedId = (id, abortCondition) => {
  return useMemo(() => {
    if (id) return id;
    if (abortCondition()) return null

    return generateId()
  }, [id])
}
```

Feels a bit too much to me. What do you think  @duranmla @snapsnapturtle @lukahartwig ?

Reply:
The idea behind the logic is the following. 

1. The user set props have the highest priority, we cannot drop them, we suppose they know what they do. Even if `id` is not needed (e.g. there is no label set), we set it anyways.
2. We generate `id` only when it must present

Reply:
But it is not like:

```js
const useGeneratedId = (id) => {
  return useMemo(() => {
    if (id) return id;
    return generateId()
  }, [id])
}
```

Why do we need the "abort condition", users will set or not set the id isn't? Nevertheless, if the abstraction is expensive I will for sure favor the duplication of a simpler code instead. 👌

Reply:
Probably it is a premature optimization, I'll do the simple version and then we'll see